topostats/tracing/dnatracing.py

```diff
@@ -493,69 +488,88 @@ def get_splined_traces(self):
                 if not self.splining_success:
                     continue
 
-                spline_average = np.divide(spline_running_total, [step_size, step_size])
+                spline_average = np.divide(spline_running_total, [step_size_px, step_size_px])
                 del spline_running_total
+                spline_average = np.delete(spline_average, -1, 0)
                 self.splined_traces[dna_num] = spline_average
-                # else:
-                #    x = self.fitted_traces[dna_num][:,0]
-                #    y = self.fitted_traces[dna_num][:,1]
-
-                #    try:
-                #        tck, u = interp.splprep([x, y], s=0, per = 2, quiet = 1, k = 3)
-                #        out = interp.splev(np.linspace(0,1,nbr*step_size), tck)
-                #        splined_trace = np.column_stack((out[0], out[1]))
-                #        self.splined_traces[dna_num] = splined_trace
-                #    except ValueError: #if the trace is really messed up just delete it
-                #        self.mol_is_circular.pop(dna_num)
-                #        self.disordered_trace.pop(dna_num)
-                #        self.grains.pop(dna_num)
-                #        self.ordered_traces.pop(dna_num)
 
             else:
-                """
-                start_x = self.fitted_traces[dna_num][0, 0]
-                end_x = self.fitted_traces[dna_num][-1, 0]
+                # ev_array = np.linspace(0, 1, 1000)
+                ev_array = np.linspace(0, 1, nbr * step_size_px)
 
-                for i in range(step_size):
-                    x_sampled = np.array([self.fitted_traces[dna_num][:, 0][j] for j in
-                                          range(i, len(self.fitted_traces[dna_num][:, 0]), step_size)])
-                    y_sampled = np.array([self.fitted_traces[dna_num][:, 1][j] for j in
-                                          range(i, len(self.fitted_traces[dna_num][:, 1]), step_size)])
-
-                    interp_f = interp.interp1d(x_sampled, y_sampled, kind='cubic', assume_sorted=False)
+                for i in range(step_size_px):
+                    x_sampled = np.array(
+                        [
+                            self.fitted_traces[dna_num][:, 0][j]
+                            for j in range(i, len(self.fitted_traces[dna_num][:, 0]), step_size_px)
```

Comment:
What is the difference between `fitted_traces` and `ordered_traces` that means they need separate handling?

(I need to check the logic elsewhere to see what is being done to `ordered_traces`.

Reply:
The idea is that, the splinining is based on `fitted_traces`, but if that gives errors, the splining is done to `ordered_traces` instead

Reply:
What errors might arise and why? If we can understand those its surely useful. Is there a reason why `fitted_traces` is preferred if it works? Is there any reason not to just use `ordered_traces` all the time?

Reply:
@ns-rse  This was from Joe's code so I am not sure what errors might arise either. `fitted_traces` are better than `ordered_traces` because they take the height of the pixels into account, instead of treating it as binary data. For a detailed explanation, see Section 2.1.5.4 of this paper: https://www.sciencedirect.com/science/article/pii/S1046202321000207